docs/src/main/asciidoc/oauth2-guide.adoc

```diff
@@ -0,0 +1,357 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Using OAuth2 RBAC
+
+include::./attributes.adoc[]
+:extension-name: Elytron Security OAuth2
+:oauth2-token: OAuth2 tokens
```

Comment:
Hmmm, what's the purpose of this one?

Reply:
I create this guide based on the JWT one and kept the variables in place, I can remove them if you want.